MainRepository/Program.cs

```diff
@@ -1,7 +1,51 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
-}
\ No newline at end of file
+﻿// This program validates the file path
+// BNF diagram of finding valid file path
+// file://C:/WorkGIT/Assignments/MainRepository/Data/bnf-diagram.PNG
+using static State;
+var strs = new List<string> { @"C:\PROGRAMS\DATA\README.TXT", @"C:\PROGRAMS\README.TXT", @"C:\PROGRAMS\DATA\DIR\README.TXT", @"C:\README.TXT", "..", "", "C:", "C:\\PROGRAMS\\README..TXT" };

```

Comment:
Such long lines should be split as shown below:
![image](https://github.com/karthick-1205/Assignments/assets/75829713/2d157115-7bbf-45ff-a2de-68a411c6b52c)